src/SerializerBuilder.php

```diff
@@ -525,8 +525,8 @@ public function build(): Serializer
 
             if (null !== $this->cacheDir) {
                 $this->createDir($this->cacheDir . '/annotations');
-                $annotationsCache = new FilesystemCache($this->cacheDir . '/annotations');
-                $annotationReader = new CachedReader($annotationReader, $annotationsCache, $this->debug);
+                $annotationsCache = new FilesystemAdapter('', 0, $this->cacheDir . '/annotations');
```

Comment:
this would result in a braking chnage for people who have not yet installed the symfony/cache component.

I would suggest to check it the `Symfony\Component\Cache\Adapter\FilesystemAdapter`  class exists use that, if not fallback to the current caching classes.